c/datasetjson.c

```diff
@@ -1671,7 +1669,117 @@ void respondWithHLQNames(HttpResponse *response, MetadataQueryCache *metadataQue
   finishResponse(response);     
 #endif /* __ZOWE_OS_ZOS */
 }
+//new function
+
+static int getDSCB(char* datasetName, char* dscb, int* dscbLength){
+  Volser volser = {0};
+  DatasetName dsn = {0};
+  memset(dsn.value, ' ', 44);
+  
+  int lParenIndex = indexOf(datasetName, strlen(datasetName),'(',0);
+  
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  if (lParenIndex > 0){
+    memcpy(dsn.value,datasetName+3,lParenIndex-3);
+  }
+  //String parsing operation, gets rid of of 3 initial unnecessary characters-> //'
+  else{
+    memcpy(dsn.value,datasetName+3,strlen(datasetName)-4);
+  }
+    
+  int volserSuccess = getVolserForDataset(&dsn, &volser);
+  
+  if(!volserSuccess){
+    
+    int rc = obtainDSCB1(dsn.value, sizeof(dsn.value),
+                           volser.value, sizeof(volser.value),
+                           dscb);
+                           
+    *dscbLength = sizeof(dscb);
```

Comment:
Value never used. Also, no point in passing it. Also, the value of this will be the size of the pointer. You can't use the sizeof function on a passed in pointer. It's passed into this function, but also passed back from obtainDSCB1, so even though it has a size and data was copied, it's still a pointer.